### ISSUE TITLE
spctl shows notarization or signature as "rejected"

### ISSUE THREAD

**Issue by fredjohnsen** [Lord-Kamina/SwiftDefaultApps#59]
I am not having a problem deploying but there does seem to be a problem with the signing:

% spctl -a -vv -t install /usr/local/bin/swda 
/usr/local/bin/swda: rejected
origin=Mac Developer: g.litenstein@gmail.com (B9BYV4VDX3)

I think it is related to this: https://developer.apple.com/documentation/xcode/using-the-latest-code-signature-format

Thanks.

**Comment by kbluck** [comment]
You can fix this locally with:
`xattr -d com.apple.quarantine /path/to/swda`

I tried screwing around with `spctl` and adding a rule didn't work, even disabling GateKeeper completely didn't work. So, something about the file having this attribute due to being downloaded. I don't know that this can be fixed in the build.

**Comment by fredjohnsen** [comment]
The quarantine flag is always added to anything that is downloaded. I don't think that's the issue.

spctl is showing as "rejected" so maybe your developer cert is expired or something in your profile info doesn't match?

**Comment by kbluck** [comment]
Yes, you're right. The code signature needs the new DER entitlements.

Until that happens, removing the quarantine extended attribute does allow you to run swda.

**Comment by Lord-Kamina** [comment]
> Yes, you're right. The code signature needs the new DER entitlements.
> 
> Until that happens, removing the quarantine extended attribute does allow you to run swda.

It's never gonna happen because this was a hobby project and I'm not paying to get a proper developer ID. Also, most of the LaunchServices API that this app is based on is deprecated and AFAIK there is no direct replacement for it as of yet. 

This is why I've essentially abandoned it for now.

**Comment by gurple** [comment]
And I just found it after my ages of blindly wandering the earth without RCDefaultApp.
